core/src/main/java/com/raphtory/internals/serialisers/DependencyFinder.java

```diff
@@ -0,0 +1,184 @@
+package com.raphtory.internals.serialisers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.BitSet;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+
+// https://stackoverflow.com/questions/19862866/find-java-class-dependencies-at-runtime
```

Comment:
This code is quite annoying but I just cannot come up with an alternative right now.

https://web.archive.org/web/20151231173739/http://asm.ow2.org/doc/tutorial-asm-2.0.html
this is an old example that shows how to use ASM visitors to lift all the dependencies of a class, I think it's worth having a look

Reply:
Well, I had a look at the alternative and it misses key things. I could probably re-implement this parser in scala but not sure it is a good use of my time as so far it seems to work pretty well...